ethvm/main.cpp

```diff
@@ -144,94 +118,58 @@ int main(int argc, char** argv)
 	Network networkName = Network::MainNetworkTest;
 	BlockHeader blockHeader; // fake block to be executed in
 	blockHeader.setGasLimit(maxBlockGasLimit());
-	blockHeader.setTimestamp(0);
 	bytes data;
 	bytes code;
 
 	Ethash::init();
 	NoProof::init();
-
-	for (int i = 1; i < argc; ++i)
-	{
-		string arg = argv[i];
-		if (arg == "-h" || arg == "--help")
-			help();
-		else if (arg == "-V" || arg == "--version")
-			version();
-		else if (arg == "--vm" && i + 1 < argc)
-		{
-			string vmKindStr = argv[++i];
-			if (vmKindStr == "interpreter")
-				vmKind = VMKind::Interpreter;
-#if ETH_EVMJIT
-			else if (vmKindStr == "jit")
-				vmKind = VMKind::JIT;
-			else if (vmKindStr == "smart")
-				vmKind = VMKind::Smart;
-#endif
-			else
-			{
-				cerr << "Unknown/unsupported VM kind: " << vmKindStr << "\n";
-				return -1;
-			}
-		}
-		else if (arg == "--mnemonics")
-			st.setShowMnemonics();
-		else if (arg == "--flat")
-			styledJson = false;
-		else if (arg == "--sender" && i + 1 < argc)
-			sender = Address(argv[++i]);
-		else if (arg == "--origin" && i + 1 < argc)
-			origin = Address(argv[++i]);
-		else if (arg == "--gas" && i + 1 < argc)
-			gas = u256(argv[++i]);
-		else if (arg == "--gas-price" && i + 1 < argc)
-			gasPrice = u256(argv[++i]);
-		else if (arg == "--author" && i + 1 < argc)
-			blockHeader.setAuthor(Address(argv[++i]));
-		else if (arg == "--number" && i + 1 < argc)
-			blockHeader.setNumber(u256(argv[++i]));
-		else if (arg == "--difficulty" && i + 1 < argc)
-			blockHeader.setDifficulty(u256(argv[++i]));
-		else if (arg == "--timestamp" && i + 1 < argc)
-			blockHeader.setTimestamp(u256(argv[++i]));
-		else if (arg == "--gas-limit" && i + 1 < argc)
-			blockHeader.setGasLimit(u256(argv[++i]).convert_to<int64_t>());
-		else if (arg == "--value" && i + 1 < argc)
-			value = u256(argv[++i]);
-		else if (arg == "--network" && i + 1 < argc)
-		{
-			string network = argv[++i];
-			if (network == "Constantinople")
-				networkName = Network::ConstantinopleTest;
-			else if (network == "Byzantium")
-				networkName = Network::ByzantiumTest;
-			else if (network == "Frontier")
-				networkName = Network::FrontierTest;
-			else if (network == "Ropsten")
-				networkName = Network::Ropsten;
-			else if (network == "Homestead")
-				networkName = Network::HomesteadTest;
-			else if (network == "Main")
-				networkName = Network::MainNetwork;
-			else
-			{
-				cerr << "Unknown network type: " << network << "\n";
-				return -1;
-			}
-		}
-		else if (arg == "stats")
+	po::options_description transactionOptions("Transaction options");
+	transactionOptions.add_options()
+			("value", po::value<string>(), "<n>  Transaction should transfer the <n> wei (default: 0).")
```

Comment:
Would it be possible make this `po::value<u256>`? And also use proper types for other options?

Reply:
Yes. It's possible. Fixed.